hdrline.c

```diff
@@ -348,7 +348,9 @@ static void index_email_date(const struct ExpandoNode *node, const struct Email
       int offset = (e->zhours * 3600 + e->zminutes * 60) * (e->zoccident ? -1 : 1);
       const time_t now = e->date_sent + offset;
       tm = mutt_date_gmtime(now);
+#ifdef HAVE_STRUCT_TM_TM_GMTOFF
       tm.tm_gmtoff = offset;
+#endif /* HAVE_STRUCT_TM_TM_GMTOFF */
```

Comment:
Just skipping setting `tm_gmtoff` sounds confusing to me. As an alternative, I suggest to fallback to using `SENT_LOCAL` if an offset cannot be set. This can be documented (`%{fmt}` behaves like `%[fmt]` on systems where `struct tm` doesn't have a `tm_gmtoff` member.

Reply:
Updated, is that close to what you had in mind @gahr?

Reply:
Yep, that works. You could simplify it even further to

```c
    case SENT_SENDER:
    {
#ifdef HAVE_STRUCT_TM_TM_GMTOFF
      int offset = (e->zhours * 3600 + e->zminutes * 60) * (e->zoccident ? -1 : 1);
      const time_t now = e->date_sent + offset;
      tm = mutt_date_gmtime(now);
      tm.tm_gmtoff = offset;
      break;
#else
      FALLTHROUGH;
#endif
    }